index.js

```diff
@@ -28,17 +28,25 @@ function filterHidden (item: string) {
 // Return a flattened-out file list of a rootDir.
 // Each entry consists of the filename (with path relative to `rootDir`).
 // DOES NOT FOLLOW SYMLINKS
-function readDirRecurse (rootDir: string, ignoreHidden: ?boolean): Promise {
+function readDirRecurse (rootDir: string, opts: Object): Promise {
   return new Promise((resolve, reject) => {
     rootDir = path.resolve(rootDir)
     const fileEntries = []
     const options = {}
-    if (ignoreHidden) {
+    if (opts && opts.ignoreHidden) {
       options.filter = filterHidden
     }
     klaw(rootDir, options)
       .on('data', (item) => {
-        if (!item.stats.isDirectory() && !item.stats.isSymbolicLink()) {
+        let excluded = false
+        if (opts && opts.excludeDirectories) {
+          for (const dir of opts.excludeDirectories) {
```

Comment:
Please rewrite this `for` block to use `Array.prototype.filter()` instead.

For example:
> excluded = opts.excludeDirectories.filter((dir) => item.path.indexOf(dir) !== -1).length > 0

Reply:
Fixed :+1: